Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateObserver.swift

```diff
@@ -101,6 +101,7 @@ extension ExperienceStateMachine {
             case let .success(.renderingStep(experience, stepIndex, _, isFirst: false)):
                 trackLifecycleEvent(.stepSeen, LifecycleEvent.properties(experience, stepIndex))
             case let .success(.endingStep(experience, stepIndex, _)):
+                trackFormCompletion(experience, stepIndex)
```

Comment:
this is causing me some uncertainty still, as it would submit a form even if the user chose to dismiss the flow - trying to see what the desired behavior really is.  Having to find more conditional ways to trigger this outside of just during step completion is also concerning though - might complicate the code a bit.

Reply:
Merging this and will update when the form completion events are triggered in a subsequent PR.